src/components/L2/Dropdowns/FeeDropdown.tsx

```diff
@@ -10,20 +10,49 @@ const PRICE_LABELS = ['Fast', 'Average', 'Slow'];
 
 export interface GasPrice {
   price: number;
-  wait: number;
+  wait: string;
 }
 
-const formatPriceWait = ({ price, wait }: GasPrice) =>
-  `${price} gwei (${Math.round(wait * 100) / 100} min)`;
+export const formatWait = (wait: number) => Math.round(wait * 100) / 100;
+export const formatDisplay = ({ price, wait }: GasPrice) =>
+  `${price} gwei (${wait} min)`;
 
-const FeeDropdown = ({ setGasPrice }: { setGasPrice: (g: number) => void }) => {
+export default function FeeDropdown({
+  setGasPrice,
+}: {
+  setGasPrice: (g: number) => void;
+}) {
   const { suggestedGasPrices } = useGasPrice(DEFAULT_GAS_PRICE_GWEI);
 
-  const [open, setOpen] = useState<boolean>(false);
+  const [open, setOpen] = useState(false);
+  const [custom, setCustom] = useState<string>('0');
   const [selected, setSelected] = useState<GasPrice>(
     suggestedGasPrices.average
   );
 
+  const handleCustom = (e: FormEvent) => {
+    const cleanedValue = e.target.value.replace(/[^0-9]/g, '');
+    const cleanedNum = Number(cleanedValue);
+    setCustom(cleanedValue);
+    setGasPrice(cleanedNum);
+
+    let customWait = 'Unknown';
+
+    const { fast, average, low } = suggestedGasPrices;
```

Comment:
Eventually this  could change to accommodate the switch to the Etherscan API on `master`

https://github.com/urbit/bridge/blob/5c8fc4f60122da759a31a61bf530fb764a727b99/src/lib/getSuggestedGasPrice.js#L14